core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -394,9 +394,15 @@ private[spark] object JsonProtocol {
 
   /** Convert executor metrics to JSON. */
   def executorMetricsToJson(executorMetrics: ExecutorMetrics): JValue = {
-    val metrics = ExecutorMetricType.values.map{ metricType =>
-      JField(metricType.name, executorMetrics.getMetricValue(metricType))
-     }
+    val metrics = for {
+      (m, _) <- ExecutorMetricType.definedMetricsAndOffset.toSeq
```

Comment:
`definedMetricsAndOffset.toSeq` causes generated JSON with a random order. Some UTs in `JsonProtocalSuite` will fail.

Reply:
Thanks for the comment. However the unit test passed even with this. The failure in the latest run wasn't related to this and it was just because of a termination. I can work on changing this though to make the order specific if that is desired.

Reply:
I am sorry it seems that there was a unit test failure after my latest change. I can produce it locally. Thank you for your report.

Reply:
I fixed this in my last commit. Sorry again